src/System.Private.CoreLib/src/System/Threading/ThreadPool.Unix.cs

```diff
@@ -22,103 +22,48 @@ public bool Unregister(WaitHandle waitObject)
 
     public static partial class ThreadPool
     {
-        // TODO: this is a very primitive (temporary) implementation of Thread Pool to allow Tasks to be
-        // used on Unix. All of this code must be replaced with proper implementation.
-
-        /// <summary>
-        /// Max allowed number of threads in the thread pool. This is just arbitrary number
-        /// that is used to prevent unbounded creation of threads.
-        /// It should by high enough to provide sufficient number of thread pool workers
-        /// in case if some threads get blocked while running user code.
-        /// </summary>
-        private static readonly int MaxThreadCount = 4 * ThreadPoolGlobals.processorCount;
-
-        /// <summary>
-        /// Semaphore that is used to release waiting thread pool workers when new work becomes available.
-        /// </summary>
-        private static SemaphoreSlim s_semaphore = new SemaphoreSlim(0);
-
-        /// <summary>
-        /// Number of worker threads created by the thread pool.
-        /// </summary>
-        private static volatile int s_workerCount = 0;
-
         public static bool SetMaxThreads(int workerThreads, int completionPortThreads)
         {
-            // Not supported at present
-            return false;
+            return ClrThreadPool.SetMaxThreads(workerThreads);
```

Comment:
These functions should still check for errors in parameters that are not used and return false before making any changes